src/app/features/home/details/actions/actions.page.ts

```diff
@@ -39,54 +44,101 @@ export class ActionsPage {
     private readonly route: ActivatedRoute,
     private readonly authService: DiaBackendAuthService,
     private readonly snackBar: MatSnackBar,
-    private readonly dialog: MatDialog
+    private readonly dialog: MatDialog,
+    private readonly storeService: DiaBackendStoreService
   ) {}
 
-  openAction(action: Action) {
+  openActionDialog$(action: Action) {
     return combineLatest([
       this.actionsService.getParams$(action.params_list_custom_param1),
       this.authService.token$,
       this.id$,
-    ])
+    ]).pipe(
+      first(),
+      concatMap(([params, token, id]) => {
+        const dialogRef = this.dialog.open<ActionsDialogComponent>(
+          ActionsDialogComponent,
+          {
+            disableClose: true,
+            data: {
+              action: action,
+              params: params,
+            },
+          }
+        );
+        return dialogRef.afterClosed().pipe(
+          isNonNullable(),
+          concatMap(data =>
+            of({
+              networkApp: action.network_app_id_text,
+              actionArgs: { ...data, token: token, cid: id },
+            } as CreateOrderInput)
+          )
+        );
+      })
+    );
+  }
+
+  openOrderDialog$(orderStatus: NetworkAppOrderStatus) {
+    const dialogRef = this.dialog.open<OrderDetailDialogComponent>(
+      OrderDetailDialogComponent,
+      {
+        disableClose: true,
+        data: orderStatus,
+        width: '80%',
+      }
+    );
+    return dialogRef.afterClosed().pipe(
+      isNonNullable(),
+      concatMap((orderId: string) => of(orderId))
+    );
+  }
+
+  createOrder$(appName: string, actionArgs: any) {
+    return this.storeService.createNetworkAppOrder(appName, actionArgs).pipe(
+      catchError((err: unknown) => {
+        return this.errorService.toastError$(err);
+      }),
+      isNonNullable()
+    );
+  }
+
+  confirmOrder$(id: string) {
+    return this.storeService.confirmNetworkAppOrder(id).pipe(
+      catchError((err: unknown) => {
+        return this.errorService.toastError$(err);
```

Comment:
Could consider adding error handling for potential errors. I don't think it's a merge blocker but it would enhance the UX.

If user want to confirm order but does not have enough num
```
{
    "error": {
        "type": "insufficient_fund",
        "message": "Insufficient fund for the order payment."
    },
    "status_code": 400
}
```

Show something like `Insufficient NUM token`